### PR TITLE
Refactored code to have separate CLI and logic

### PR DESCRIPTION
**Pull request by eirannejad**
- rhino-agnostic cli builder and logic module
- added build script for rhino
- multiple targets

### CODE REVIEW

**Comment by eirannejad** [comment]
- Refactored the code to separate CLI from logic. The logic dlls are now used with pyRevit on Revit to generate stubs from the running instance of Revit and for that specific API version
- Added x86 and x64 targets for both the cli and the logic module
- Added powershell build script for default Rhino stubs
- Updated .gitignore

CLI usage
```
Usage:
    pystubsbuilder (-h | --help)
    pystubsbuilder (-V | --version)
    pystubsbuilder [--dest=<dest_path>] [--search=<search_path>...] <target_dll>...
```

Accepts destination path `--dest`
Accepts multiple search paths `--search`
Accepts multiple target dlls

**Comment by sbaer** [comment]
If I run `build-rhino-stubs.ps1`, the resulting stubs end up with a slightly different path. I would fix this, but I don't want to mess up expected paths for your Revit project

**Comment by eirannejad** [comment]
Whoops you are right. I'll revise to support both methods. Great catch

**Comment by eirannejad** [comment]
Added more options to control the output dir structure

```
    --prefix=<prefix>           Root namespace directory prefix
    --postfix=<postfix>         Root namespace directory postfix
    --dest-is-root              Use destination path for root namespace
```

`--prefix` and `--postfix` can extend the final root namespace directory name.
`--dest-is-root` is to force the utility to use the `--dest` path as the root namespace directory

**Comment by sbaer** [comment]
Hmm.. the paths are correct, but for some reason the generated .pyi files are being created with contents in a different order. That may be fine, but I need more time to review

**Comment by sbaer** [comment]
Something is definitely different in the stubs generation logic. If I roll back to master, I can see that the stubs are getting generated the same way they used to be getting generated.

**Comment by eirannejad** [comment]
Seems like the order is different. I didn't mess with the stub generator code. I think we need to add ordering for consistent output

File: builder/PythonStubsBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    public static class PythonStubsBuilder {
+        private static List<string> SearchPaths { get; set; } = new List<string>();
+
+        public static string BuildAssemblyStubs(string targetAssemblyPath, string destPath = null, string[] searchPaths = null, PythonStubsBuilderConfigs cfgs = null) {
+            // prepare configs
+            if (cfgs is null)
+                cfgs = new PythonStubsBuilderConfigs();
+
+            // prepare resolver
+            AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve; ;
```
Comment (sbaer) [inline builder/PythonStubsBuilder.cs:22]: We're adding the same AssemblyResolver over and over every time this function is called.

Comment (eirannejad) [reply builder/PythonStubsBuilder.cs:22]: Should I add a static bool flag to mark "AlreadyHandling" or there is a better way?

```csharp
if (!alreadyHandling)
    AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve;
```

Comment (sbaer) [reply builder/PythonStubsBuilder.cs:22]: You could also just always do a `-=` before a `+=` call

**Pull request merged**